src/components/DeliverableView/DocumentsUploader.tsx

```diff
@@ -23,21 +38,29 @@ const DocumentsUploader = ({ deliverable }: ViewProps): JSX.Element => {
 
   const onCloseFileUploadDialog = useCallback(() => void setFiles([]), []);
 
+  useEffect(() => {
+    if (!canEdit && documentLimitReached && showDocumentLimitReachedMessage) {
+      showDocumentLimitReachedMessage();
+    }
+  }, [canEdit, documentLimitReached, showDocumentLimitReachedMessage]);
+
   return (
     <Box display='flex' flexDirection='column'>
       {files.length > 0 && (
         <FileUploadDialog deliverable={deliverable} files={files} onClose={onCloseFileUploadDialog} />
       )}
-      <FileChooser
-        acceptFileType='image/*,application/*'
-        chooseFileText={strings.CHOOSE_FILE}
-        setFiles={setFiles}
-        maxFiles={15 - deliverable.documents.length}
-        multipleSelection
-        uploadDescription={strings.UPLOAD_FILES_DESCRIPTION}
-        uploadText={strings.UPLOAD_FILES_TITLE}
-        template={template}
-      />
+      {!documentLimitReached && (
+        <FileChooser
+          acceptFileType='image/*,application/*'
+          chooseFileText={strings.CHOOSE_FILE}
+          maxFiles={canEdit ? undefined : MAX_FILES_LIMIT - deliverable.documents.length}
```

Comment:
We could pass the max files into the DocumentsUploader component as an optional prop and remove the canEdit prop.
Technically both participants and global-roles with edit permissions can upload (edit) files.